ONSdigital/census-rm-qid-batch-runner
# 728 rmc 319b generate ce spg initial contact letter unaddressed print files ew

# Motivation and Context

For Census 2021 we're going to have to create unaddressed CE/SPG initial contact letters for QM and PPO. To be able to do this, qid-Batch-runner had to be updated to be able to cope with sending batches to different locations. This has been done by splitting the batches in suppliers and types and updating the code so when we pass in the PPO supplier, it'll put the batch into the correct SFTP directory.

# What has changed



- Now able to pass in a supplier when generating print files to decide which sftp location the files will go to.
- Added mappings for suppliers to keys and sftp directories.
- Added updated tests
# How to test?



- Run `make test` and all should pass.
- Generate a PPO print files and pass in the PPO supplier. It should generate files into the ppo_dev sftp directory.

# Links



[Trello](https://trello.com/c/Ij81s0mo/)
